unstructured/ingest/connector/salesforce.py

```diff
@@ -0,0 +1,328 @@
+"""
+Salesforce Connector
+Able to download Account, Case, Campaign, EmailMessage, Lead,
+Salesforce returns everything as a list of json.
+This saves each entry as a separate file to be partitioned.
+Using JWT authorization
+https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_auth_key_and_cert.htm
+https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_auth_connected_app.htm
+"""
+import os
+from dataclasses import dataclass
+from email.utils import formatdate
+from pathlib import Path
+from string import Template
+from textwrap import dedent
+from typing import Any, Dict, List, Type
+
+from dateutil import parser  # type: ignore
+from simple_salesforce import Salesforce
+from simple_salesforce.exceptions import SalesforceMalformedRequest
```

Comment:
have been changing elsewhere to make these import inline so that request_dependencies can properly catch and handle this. 

that said, @rbiseck3 is working on a change to wrap the runner instead. so if you want to wait until that merges then we can just leave this as is.

Reply:
@ryannikolaidis you're talking about moving the imports to where there are used and not up top? Or are you talking about putting both imports in 1 line?

Reply:
I see from another PR that you want it where the imports are actually used.

Reply:
sorry, this got buried on my side. yes, exactly, import in the functions where they are used or where the exceptions are raised. then wrap with the request_dependencies decorator and then the user will get the `pip install unstructured[salesforce]` message rather than complaint about single dependency that is missing.

Reply:
Cool. Makes sense. I think you meant `requires_dependencies` above instead of `request_dependencies`. That threw me a little bit.

I put the Exception import in the function.

Reply:
oh yep, sorry! awesome, thanks.